File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        output = [0]*(n+1)
        for i in range(n+1):
            res = 0
            temp = i
            while temp:
                temp = temp & (temp - 1)
                res += 1
            output[i] = res
        return output

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        // start from 0, and expand window
        // has k chance to replace.
        // what if abbbbb
        unordered_map<char,int> temp;
        int l = 0, r = 0;
        int output = 0;
        int maxi = 0;
        while(l<=r && r<s.size()){
            temp[s[r]]++;
            if(temp[s[r]] > maxi){
                maxi = temp[s[r]];
            }
            while((r-l+1)-maxi>k){
                temp[s[l++]]--;
            }
            if(output < r-l+1){
                output = r-l+1;
            }
            r++;
        }
        return output;
        // was thinking how to search max if l-- was max and check max again, but
        // dont need to do that



    }
};
